internal/parse/parser_test.go

```diff
@@ -171,106 +130,187 @@ func TestRound(t *testing.T) {
+				"FROM person AS p INNER JOIN address AS a ON p.address_id =] " +
+				"InputPart[Address.ID] " +
+				"BypassPart[WHERE p.name =] " +
+				"InputPart[Person.name] " +
+				"BypassPart[AND p.address_id =] " +
+				"InputPart[Person.address_id]]",
 		},
 		{
 			"SELECT p.*, a.district " +
 				"FROM person AS p JOIN address AS a ON p.address_id = a.id " +
 				"WHERE p.name = $Person.*",
-			"ParsedExpr[BypassPart[SELECT p.*, a.district FROM person AS p JOIN address AS a ON p.address_id = a.id WHERE p.name = $Person.*]]",
-			[]any{&Person{}},
-			[]any{&Person{}},
-			"SELECT p.*, a.district FROM person AS p JOIN address AS a ON p.address_id = a.id WHERE p.name = ?",
+			"ParsedExpr[BypassPart[SELECT p.*, a.district FROM person AS p JOIN address AS a ON p.address_id = a.id WHERE p.name =] " +
+				"InputPart[Person.*]]",
 		},
 		{
 			"INSERT INTO person (name) VALUES $Person.name",
-			"ParsedExpr[BypassPart[INSERT INTO person (name) VALUES $Person.name]]",
-			[]any{&Person{}},
-			[]any{&Person{}},
-			"INSERT INTO person (name) VALUES ?",
+			"ParsedExpr[BypassPart[INSERT INTO person (name) VALUES] " +
+				"InputPart[Person.name]]",
 		},
 		{
 			"INSERT INTO person VALUES $Person.*",
-			"ParsedExpr[BypassPart[INSERT INTO person VALUES $Person.*]]",
-			[]any{&Person{}},
-			[]any{&Person{}},
-			"INSERT INTO person VALUES ?",
+			"ParsedExpr[BypassPart[INSERT INTO person VALUES] " +
+				"InputPart[Person.*]]",
 		},
 		{
 			"UPDATE person SET person.address_id = $Address.ID " +
 				"WHERE person.id = $Person.ID",
-			"ParsedExpr[BypassPart[UPDATE person SET person.address_id = $Address.ID " +
-				"WHERE person.id = $Person.ID]]",
-			[]any{&Address{}, &Person{}},
-			[]any{&Address{}, &Person{}},
-			"UPDATE person SET person.address_id = ?  WHERE person.id = ?",
+			"ParsedExpr[BypassPart[UPDATE person SET person.address_id =] " +
+				"InputPart[Address.ID] " +
+				"BypassPart[WHERE person.id =] " +
+				"InputPart[Person.ID]]",
 		},
 	}
 
 	parser := parse.NewParser()
 	for i, test := range tests {
 		var parsedExpr *parse.ParsedExpr
-		if parsedExpr, _ = parser.Parse(test.input); parsedExpr.String() !=
-			test.expectedParsed {
-			t.Errorf("Test %d Failed (Parse): input: %s\nexpected: %s\nactual:   %s\n", i, test.input, test.expectedParsed, parsedExpr.String())
+		var err error
+		if parsedExpr, err = parser.Parse(test.input); err != nil {
+			t.Errorf("test %d failed (Parse): input: %s\nexpected: %s\nerr: %s\n", i, test.input, test.expectedParsed, err)
+		} else if parsedExpr.String() != test.expectedParsed {
+			t.Errorf("test %d failed (Parse): input: %s\nexpected: %s\nactual:   %s\n", i, test.input, test.expectedParsed, parsedExpr.String())
 		}
 	}
 }
+
+// We return a proper error when we find an unbound string literal
+func TestUnfinishedStringLiteral(t *testing.T) {
+	sql := "select foo from t where x = 'dddd"
```

Comment:
Per conversation in the sprint, can we please standardize on using casing such as `SELECT foo FROM` to make the statements more readable?  This is already the case in the tests above.

Reply:
Standardised.